sounddevice.py

```diff
@@ -2289,14 +2289,40 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    error_message : str, optional
+        User error message.
+    pa_error_code : int, optional
+        PaErrorCode value.
+    hosterror_info : tuple, optional
+        3-tuple containing the host API index, host error code, and the host
+        error message.
+
+    """
+    def __str__(self):
+        errormsg = self.args[0] if len(self.args) > 0 else ''
+
+        if len(self.args) > 2:
+            host_api, hosterror_code, hosterror_text = self.args[2]
+            hostname = query_hostapis(host_api)['name']
+            errormsg += " [PaErrorCode {0}, {1} error {2}]"\
+                        .format(self.args[1], hostname, hosterror_code)
+            if hosterror_text:
+                errormsg += ": '{0}'".format(hosterror_text)
```

Comment:
And wouldn't it be better to read if the error number(s) were in the very end?

    PortAudioError: Error opening Stream: Unanticipated host error: 'Invalid argument' [PaErrorCode -9999, ALSA error -22]